src/coreclr/nativeaot/Runtime/unix/unixasmmacrosarm64.inc

```diff
@@ -237,6 +237,17 @@ C_FUNC(\Name):
     EPILOG_RESTORE_REG_PAIR_INDEXED   fp, lr, 32
 .endm
 
+.macro GETTHREAD_ETLS_2
+    PROLOG_SAVE_REG_PAIR_INDEXED   fp, lr, -32           // ;; Push down stack pointer and store FP and LR
+    stp         x0, x1, [sp, #0x10]
```

Comment:
This needs to preserve the floating-point return registers too. It is not 100% safe to depend on `RhpGetThread` to not corrupt them.

Reply:
We need to make sure everything that `PUSH_PROBE_FRAME` stores is preserved around call to `RhpGetThread`. 

Assuming that the call will take case of calee-saved state, we only need to save integer returns {x0, x1} and FP/HFA/HVA  {q0, q1, q2, q3}   
It is unlikely that `RhpGetThread` will use them, but it is not guaranteed.

Reply:
I think same applies to other ETLS  macros too.

I see an old comment at ETLS_9 use 

```c
#ifdef FEATURE_EMULATED_TLS
        // This doesn't save and restore the floating point argument registers. If we encounter a
        // target system that uses TLS emulation and modify these registers during this call we
        // need to save and restore them, too
        GETTHUNKDATA_ETLS_9
```

it looks like not saving FP was intentional at some point.

Reply:
Ok, I guess that we can continue with this assumption then.